internal/cli/atlas/deployments/options/deployment_opts_telemetry.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package options
+
+import (
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/telemetry"
+)
+
+//go:generate mockgen -destination=../../../../mocks/mock_deployment_opts_telemetry.go -package=mocks github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/atlas/deployments/options DeploymentTelemetry
+type DeploymentTelemetry interface {
+	AppendDeploymentType()
+}
+
+func NewDeploymentTypeTelemetry(opts *DeploymentOpts) DeploymentTelemetry {
```

Comment:
[nit] in go the r[ecommendation is to accept interfaces and return structs](https://tutorialedge.net/golang/accept-interfaces-return-structs/), not the other way around

Reply:
we can do that it may look a bit weird for `DeploymentOpts` though as there we need to use `DeploymentType` we'd need to cast with something like this (which works fine):

```shell
func NewDeploymentTypeTelemetry(opts DeploymentTelemetry) *DeploymentOpts {
	return opts.(*DeploymentOpts)
}
```

Reply:
it's a bit weird that you need to cast it to an interface if it accepts an interface but let's leave this for now

Reply:
I think the problem may be this is not really a "new" which should return a new object but this is more of a "cast as interface" thing